mesonbuild/programs.py

```diff
@@ -55,6 +55,12 @@ def __init__(self, name: str, command: T.Optional[T.List[str]] = None,
                 if ret:
                     self.command = ret + args
                 else:
+                    if os.path.isabs(cmd) and not os.path.exists(cmd):
+                        for ext in self.windows_exts:
+                            trial_ext = f'{cmd}.{ext}'
+                            if os.path.exists(trial_ext):
+                                cmd = trial_ext
+                                break
```

Comment:
Since this is copied from search_windows_special_cases you should include the comment from there as well.